SmartTutor/ContentModel/SubscriptionService.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using SmartTutor.ContentModel.Exceptions;
+using SmartTutor.ContentModel.Lectures;
+using SmartTutor.ContentModel.Subscriptions;
+using SmartTutor.ContentModel.Subscriptions.Repository;
+using SmartTutor.Controllers.Content.DTOs;
+
+namespace SmartTutor.ContentModel
+{
+    public class SubscriptionService : ISubscriptionService
+    {
+        private readonly ISubscriptionRepository _subscriptionRepository;
+
+        public SubscriptionService(ISubscriptionRepository subscriptionRepository)
+        {
+            _subscriptionRepository = subscriptionRepository;
+        }
+
+        public void SubscribeTeacher(int teacherId, int individualPlanId)
+        {
+            var teacher = _subscriptionRepository.GetTeacher(teacherId);
+
+            if (teacher.GetActiveSubscription() != null)
+                throw new TeacherAlreadySubscribedException(teacherId.ToString());
+
+            var individualPlanUsage = new IndividualPlanUsage(individualPlanId);
+            var individualPlan = _subscriptionRepository.GetIndividualPlan(individualPlanId);
+            individualPlanUsage = _subscriptionRepository.SaveOrUpdatePlanUsage(individualPlanUsage);
+            var end = GetEndDate(individualPlan);
```

Comment:
Can we transform this into individualPlan.GetEndDate(). We can remove the end variable and simply create a subscription using individualPlan.GetEndDate() inline.